src/Resources/ResourceManager/Implementation/DeploymentScripts/SaveAzDeploymentScriptLog.cs

```diff
@@ -0,0 +1,136 @@
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
+{
+    [Cmdlet(VerbsData.Save, AzureRMConstants.AzureRMPrefix + "DeploymentScriptLog", SupportsShouldProcess = true, DefaultParameterSetName = SaveAzDeploymentScriptLog.SaveDeploymentScriptLogByName), OutputType(typeof(PsDeploymentScriptLogPath))]
+
+    public class SaveAzDeploymentScriptLog : DeploymentScriptCmdletBase
+    {
+        #region Cmdlet Parameters and Parameter Set Definitions
+
+        internal const string SaveDeploymentScriptLogByName = "SaveDeploymentScriptLogByName";
+        internal const string SaveDeploymentScriptLogByResourceId = "SaveDeploymentScriptLogByResourceId";
+        internal const string SaveDeploymentScriptLogByInputObject = "SaveDeploymentScriptLogByInputObject";
+
+        [Parameter(Position = 0, ParameterSetName = SaveDeploymentScriptLogByName, Mandatory = true,
+            ValueFromPipelineByPropertyName = true, HelpMessage = "The name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Position = 1, ParameterSetName = SaveDeploymentScriptLogByName, Mandatory = true,
+            ValueFromPipelineByPropertyName = true, HelpMessage = "The name of the deployment script.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Position = 0, ParameterSetName = SaveDeploymentScriptLogByResourceId, Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage =
+                "The fully qualified resource Id of the deployment script. Example: /subscriptions/{subId}/resourceGroups/{rgName}/providers/Microsoft.Resources/deploymentScripts/{deploymentScriptName}")]
+        [ValidateNotNullOrEmpty]
+        [ResourceIdCompleter("Microsoft.Resources/deploymentScripts")]
+        public string DeploymentScriptResourceId { get; set; }
+
+        [Parameter(Position = 0, ParameterSetName = SaveDeploymentScriptLogByInputObject, Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The deployment script PowerShell object.")]
+        [ValidateNotNullOrEmpty]
+        public PsDeploymentScript DeploymentScriptInputObject { get; set; }
+
+        [Parameter(Position = 2, ParameterSetName = SaveDeploymentScriptLogByName, Mandatory = true,
+            ValueFromPipelineByPropertyName = true, HelpMessage = "The directory path to save deployment script log.")]
+        [Parameter(Position = 1, ParameterSetName = SaveDeploymentScriptLogByResourceId, Mandatory = true,
+            ValueFromPipelineByPropertyName = true, HelpMessage = "The directory path to save deployment script log.")]
+        [Parameter(Position = 1, ParameterSetName = SaveDeploymentScriptLogByInputObject, Mandatory = true,
+            ValueFromPipelineByPropertyName = true, HelpMessage = "The directory path to save deployment script log.")]
```

Comment:
There isn't gonna be a piping scenario that the output of the first cmdlet contains a property named "OutputPath", so it's not necessary to use `ValueFromPipelineByPropertyName`.
```suggestion
        [Parameter(Position = 2, ParameterSetName = SaveDeploymentScriptLogByName, Mandatory = true,
            HelpMessage = "The directory path to save deployment script log.")]
        [Parameter(Position = 1, ParameterSetName = SaveDeploymentScriptLogByResourceId, Mandatory = true,
            HelpMessage = "The directory path to save deployment script log.")]
        [Parameter(Position = 1, ParameterSetName = SaveDeploymentScriptLogByInputObject, Mandatory = true,
            HelpMessage = "The directory path to save deployment script log.")]
```

Reply:
Updated.

---
In reply to: [392791889](https://github.com/Azure/azure-powershell/pull/11266#discussion_r392791889) [](ancestors = 392791889)